src/syscalls/tasks.rs

```diff
@@ -289,19 +289,25 @@ pub extern "C" fn sys_join(id: Tid) -> i32 {
 /// Mapping between TaskID and TaskHandle
 static TASKS: SpinlockIrqSave<BTreeMap<TaskId, TaskHandle>> = SpinlockIrqSave::new(BTreeMap::new());
 
-fn __sys_block_current_task() {
+fn __sys_block_current_task(timeout: Option<u64>) {
 	let core_scheduler = core_scheduler();
 	let handle = core_scheduler.get_current_task_handle();
 	let tid = core_scheduler.get_current_task_id();
 
 	TASKS.lock().insert(tid, handle);
-	core_scheduler.block_current_task(None);
+	core_scheduler.block_current_task(timeout);
```

Comment:
`block_current_task` takes an absolute timestamp (`Instant`) instead of a relative time (`Duration`), doesn't it?

So currently this would result in basically no sleeping at all. If we were to replace `sys_block_current_task_with_timeout` (see below), we could avoid this altogether.